src/main/java/racing/domain/car/Cars.java

```diff
@@ -0,0 +1,80 @@
+package racing.domain.car;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import racing.domain.startegy.MovingStrategy;
+
+public class Cars {
+
+    private static final int RANGE_OF_RANDOM_NUMBER = 10;
+
+    private final List<Car> cars;
+
+    public Cars(List<Car> cars) {
+        this.cars = cars;
+        checkZeroCar(this.cars);
+        checkSameNameCars(this.cars);
+    }
+
+    public Cars(List<String> names, int initValue) {
+        this(
+            names.stream()
+                .map(name -> new Car(name, initValue))
+                .collect(Collectors.toList())
+        );
+    }
+
+    private void checkSameNameCars(List<Car> cars) {
+        if (cars.size()
+            != cars.stream().map(Car::getNameValue).distinct().count()) {
```

Comment:
stream 문을 if문에 쭉 대입하기 보다는 적절히 변수로 추출하여 사용하는 것이 가독성에 더 도움이 됩니다. 😃 

```suggestion
        long distinctCarsSize = cars.stream()
                .map(Car::getNameValue)
                .distinct()
                .count();
        if (cars.size() != distinctCarsSize) {
```

Reply:
오! 이렇게 긴 stream 문장을 변수에 넣고, 이를 사용하니 확실히 가독성이 좋아지네요! 감사합니다!